tests/test_acl.py

```diff
@@ -1,417 +1,240 @@
-        keys = self.asic_db.wait_for_n_keys("ASIC_STATE:SAI_OBJECT_TYPE_ACL_ENTRY", num_keys)
-
-        acl_entries = [k for k in keys if k not in self.asic_db.default_acl_entries]
-        for entry in acl_entries:
-            rule = self.asic_db.wait_for_entry("ASIC_STATE:SAI_OBJECT_TYPE_ACL_ENTRY", entry)
-            priority = rule.get("SAI_ACL_ENTRY_ATTR_PRIORITY", None)
-            assert priority in priorities
-            self._check_acl_entry(rule, expected[priority],
-                                  action=in_actions[priority], priority=priority)
-
-    def _check_acl_entry(self, entry, qualifiers, action, priority):
-        acl_table_id = self.get_acl_table_id()
-
-        for k, v in entry.items():
-            if k == "SAI_ACL_ENTRY_ATTR_TABLE_ID":
-                assert v == acl_table_id
-            elif k == "SAI_ACL_ENTRY_ATTR_ADMIN_STATE":
-                assert v == "true"
-            elif k == "SAI_ACL_ENTRY_ATTR_PRIORITY":
-                assert v == priority
-            elif k == "SAI_ACL_ENTRY_ATTR_ACTION_COUNTER":
-                assert True
-            elif k == "SAI_ACL_ENTRY_ATTR_ACTION_PACKET_ACTION":
-                if action == "FORWARD":
-                    assert v == "SAI_PACKET_ACTION_FORWARD"
-                elif action == "DROP":
-                    assert v == "SAI_PACKET_ACTION_DROP"
-                else:
-                    assert False
-            elif k == "SAI_ACL_ENTRY_ATTR_ACTION_REDIRECT":
-                if "REDIRECT" not in action:
-                    assert False
-            elif k in qualifiers:
-                assert qualifiers[k](v)
-            else:
-                assert False
-
-    def get_simple_qualifier_comparator(self, expected_qualifier):
-        def _match_qualifier(sai_qualifier):
-            return expected_qualifier == sai_qualifier
-
-        return _match_qualifier
-
-    def get_port_list_comparator(self, expected_ports):
-        def _match_port_list(sai_port_list):
-            if not sai_port_list.startswith("{}:".format(len(expected_ports))):
-                return False
-            for port in expected_ports:
-                if self.asic_db.port_name_map[port] not in sai_port_list:
-                    return False
-
-            return True
-
-        return _match_port_list
-
-    def get_acl_range_comparator(self, expected_type, expected_ports):
-        def _match_acl_range(sai_acl_range):
-            range_id = sai_acl_range.split(":", 1)[1]
-            fvs = self.asic_db.wait_for_entry("ASIC_STATE:SAI_OBJECT_TYPE_ACL_RANGE", range_id)
-            for k, v in fvs.items():
-                if k == "SAI_ACL_RANGE_ATTR_TYPE" and v == expected_type:
-                    continue
-                elif k == "SAI_ACL_RANGE_ATTR_LIMIT" and v == expected_ports:
-                    continue
-                else:
-                    return False
-
-            return True
-
-        return _match_acl_range
+    def setup_dvs_acl(self, dvs):
```

Comment:
I think we can probably get rid of BaseTestAcl at this point since we've moved all the helpers and just provide a fixture to access dvs_acl.

Reply:
👍